find-mmc-bypartlabel

```diff
@@ -34,18 +34,21 @@ while [ ! -d /sys/class/block/mmcblk0p* ] && [ ! -d /sys/class/block/sda ] ; do
     echo "find-mmc-bypartlabel: Waiting for /sys/class/block/mmcblk0p*..." > /dev/kmsg
     i=$(( ${i:-0} + 1 ))
     sleep 0.5
-    if [ $i = 10 ]; then
+    if [ $i -eq 10 ]; then
         echo "find-mmc-bypartlabel: Error: timeout waiting for /sys/class/block/mmcblk0p*" > /dev/kmsg
         exit 1
     fi
 done
-
-for mmc_sysfs in /sys/class/block/mmcblk0p*/ /sys/class/block/sda*/; do
-	if grep -q -w PARTNAME="$1" "$mmc_sysfs"/uevent 2> /dev/null; then
-		FIMAGE_DEV_NAME=$(echo $mmc_sysfs | cut -d "/" -f 5)
-		echo "/dev/$FIMAGE_DEV_NAME"
-		exit 0
-	fi
+while [ ${i:-0} -le 50 ] ; do
+	for mmc_sysfs in /sys/class/block/mmcblk0p*/ /sys/class/block/sda*/; do
+		if grep -q -w PARTNAME="$1" "$mmc_sysfs"/uevent 2> /dev/null; then
```

Comment:
I started to ponder that should the while loop above be actually looking for the /uevent file istead of the directory with -d?

So the directory is created by the /uevent is not populated yet at that point?

Reply:
This wait for the the exact file matching to the requested to be available, eg. that with the requested label.

Reply:
So if a partitions is found with file/uevent it doesn't need to be the requested partition, that's why the check is here.